File: Algoritmos-Fundamentais/Java/maximoDivisorComum.java
import java.util.Scanner;

public class maximoDivisorComum {
    public static int mdc(int a, int b) {
        a = Math.abs(a);
        b = Math.abs(b);

        while (b != 0) {
            int resto = a % b;
            a = b;
            b = resto;
        }

        return a;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o primeiro inteiro: ");
        int a = scanner.nextInt();

        System.out.print("Digite o segundo inteiro: ");
        int b = scanner.nextInt();

        int resultado = mdc(a, b);

        System.out.println("O MDC de " + a + " e " + b + " é: " + resultado);

        scanner.close();
    }
}

File: Algoritmos-Fundamentais/Javascript/maximoDivisorComum.js
function maximoDivisorComum(a, b) {
    a = Math.abs(a);
    b = Math.abs(b);

    while (b !== 0) {
        let resto = a % b;
        a = b;
        b = resto;
    }

    return a;
}

let a = parseInt(prompt("Digite o primeiro número: "));
let b = parseInt(prompt("Digite o segundo número: "));

console.log("O máximo divisor comum de " + a + " e " + b + " é: " + maximoDivisorComum(a, b));